app/scripts/models/SocketConnection.js

```diff
@@ -85,6 +94,31 @@
             this.socket = socket;
         },
 
+        onWorkerMessage: function (e) {
```

Comment:
Here we collect all parts of geometry and when it is done - trigger 'geometry-data-received:event'